packages/lib/src/components/MBWay/components/MBWayInput/MBWayInput.tsx

```diff
@@ -16,46 +15,23 @@ function MBWayInput(props: MBWayInputProps) {
 
     const { allowedCountries = [] } = props;
 
-    const [phonePrefixes, setPhonePrefixes] = useState<DataSet>([]);
-    const [status, setStatus] = useState<string>('loading');
+    const [status, setStatus] = useState<string>('ready');
 
     this.setStatus = setStatus;
     this.showValidation = phoneInputRef?.current?.triggerValidation;
 
-    useLayoutEffect(() => {
-        getDataset('phonenumbers', loadingContext)
-            .then(response => {
-                const countriesFilter = country => allowedCountries.includes(country.id);
-                const filteredCountries = allowedCountries.length ? response.filter(countriesFilter) : response;
-                const mappedCountries = filteredCountries.map(item => {
-                    // Get country flags (magic! - shifts the country code characters to the correct position of the emoji in the unicode space)
-                    const codePoints: number[] = item.id
-                        .toUpperCase()
-                        .split('')
-                        .map(char => 127397 + char.charCodeAt(0));
+    const { loadingStatus: prefixLoadingStatus, phonePrefixes, error } = usePhonePrefixes({ allowedCountries, loadingContext });
 
-                    // Get flag emoji + space at end (it doesn't work to add spaces in the template literal, below)
-                    const flag = String.fromCodePoint ? String.fromCodePoint(...codePoints) + '\u00A0\u00A0' : '';
-
-                    return { id: item.prefix, name: `${flag} ${item.prefix} (${item.id})`, selectedOptionName: `${flag} ${item.prefix}` };
-                });
-
-                setPhonePrefixes(mappedCountries || []);
-                setStatus('ready');
-            })
-            .catch(error => {
-                props.onError(new AdyenCheckoutError('ERROR', error));
-                setPhonePrefixes([]);
-                setStatus('ready');
-            });
-    }, []);
+    if (error) {
+        props.onError(new AdyenCheckoutError('ERROR', error));
+    }
```

Comment:
This part should be placed as you did before: inside the try-catch block. Otherwise, every time that component re-render, we will trigger the `onError` (if there is an error).

You can pass the error to the hook and do it there on its try-catch block:
```javascript
const { loadingStatus: prefixLoadingStatus, phonePrefixes } = usePhonePrefixes({ allowedCountries, loadingContext, handleError: props.onError });`
```

An alternative without passing the handler to the custom hook is to react using useEffect:

```javascript
useEffect(() => {
   if (error) props.onError(...);
}, [error]);
```

Reply:
alright